### PR TITLE
document arrayContainsElement filter

### PR DESCRIPTION
**Pull request by clintropolis**
Adds docs for new native filter added in #15366

### CODE REVIEW

File: docs/querying/filters.md [hunk 1]
```diff
@@ -450,6 +450,43 @@ Note that it is often more optimal to use a like filter instead of a regex for s
...
 ```
 
+## Array contains element filter
+
+The `arrayContainsElement` filter is for checking if an `ARRAY` contains a specific element, but can match against any type of column with scalar columns being treated as single element arrays.
+
+| Property | Description | Required |
+| -------- | ----------- | -------- |
+| `type` | Must be "arrayContainsElement".| Yes |
+| `column` | Input column or virtual column name to filter. | Yes |
```
[the remark this answers is not in the document]
Comment (clintropolis) [reply docs/querying/filters.md:460]: this matches all of the other tables in the page.. i guess i can update all of the tables... again :p

File: docs/querying/filters.md [hunk 2]
```diff
@@ -450,6 +450,43 @@ Note that it is often more optimal to use a like filter instead of a regex for s
...
+| `elementMatchValueType` | String specifying the type of element value to match. For example `STRING`, `LONG`, `DOUBLE`, `FLOAT`, `ARRAY<STRING>`, `ARRAY<LONG>`, or any other Druid type. The `elementMatchValueType` determines how Druid interprets the `elementMatchValue` to assist in converting to the type of elements contained in the matched `column`. | Yes |
+| `elementMatchValue` | Array element value to match, including null. | Yes |
+
+### Example: equivalent of `WHERE ARRAY_CONTAINS(someArrayColumn, 'hello')`
```
[the remark this answers is not in the document]
Comment (clintropolis) [reply docs/querying/filters.md:464]: all of the other examples on this page are framed in terms of SQL queries, am going to leave it as is

[the remark this answers is not in the document]
Comment (abhishekagarwal87) [reply docs/querying/filters.md:464]: hehe. that's what I thought but was too lazy to look beyond the surrounding functions.

File: docs/querying/filters.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (abhishekagarwal87) [reply docs/querying/filters.md:460]: its ok even if we make an exception here :)

File: docs/querying/filters.md [hunk 3]
```diff
@@ -450,6 +450,43 @@ Note that it is often more optimal to use a like filter instead of a regex for s
...
+## Array contains element filter
+
+The `arrayContainsElement` filter is for checking if an `ARRAY` contains a specific element, but can match against any type of column with scalar columns being treated as single element arrays.
+
+| Property | Description | Required |
+| -------- | ----------- | -------- |
+| `type` | Must be "arrayContainsElement".| Yes |
+| `column` | Input column or virtual column name to filter. | Yes |
+| `elementMatchValueType` | String specifying the type of element value to match. For example `STRING`, `LONG`, `DOUBLE`, `FLOAT`, `ARRAY<STRING>`, `ARRAY<LONG>`, or any other Druid type. The `elementMatchValueType` determines how Druid interprets the `elementMatchValue` to assist in converting to the type of elements contained in the matched `column`. | Yes |
+| `elementMatchValue` | Array element value to match, including null. | Yes |
```
[the remark this answers is not in the document]
Comment (317brian) [reply docs/querying/filters.md:462]: The original wording is mostly ok imo, but the explicit can be null is helpful.

```suggestion
| `elementMatchValue` | Array element value to match. This value can be null. | Yes |
```

**Review by 317brian — APPROVED** [review]
LGTM just some minor suggestions on top of Abhishek's suggestions.

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**